backend/src/apiserver/server/pipeline_server.go

```diff
@@ -985,66 +972,56 @@ func (s *PipelineServer) GetPipelineVersionTemplate(ctx context.Context, request
 }
 
 // Checks if a user can access a pipeline version.
+// Adds namespace of the parent pipeline if version id is not empty,
+// API group, version, and resource type.
 func (s *PipelineServer) canAccessPipelineVersion(ctx context.Context, versionId string, resourceAttributes *authorizationv1.ResourceAttributes) error {
 	if !common.IsMultiUserMode() {
 		// Skip authorization if not multi-user mode.
 		return nil
 	}
+	pipelineId := ""
 	if versionId != "" {
-		namespace, err := s.resourceManager.FetchNamespaceFromPipelineVersionId(versionId)
+		pipelineVersion, err := s.resourceManager.GetPipelineVersion(versionId)
 		if err != nil {
-			return util.Wrapf(err, "Failed to access pipeline version %s. Check if it exists and have a namespace assigned", versionId)
+			return util.Wrapf(err, "Failed to access pipeline version %s. Check if it exists", versionId)
 		}
-		// Allow users to access pipelines in the default namespace.
-		if s.resourceManager.IsEmptyNamespace(namespace) {
-			return nil
-		}
-		resourceAttributes.Namespace = namespace
-	}
-	err := s.checkAccess(ctx, resourceAttributes)
-	if err != nil {
-		return util.Wrapf(err, "Failed to access pipeline version %s. Check if you have access to namespace %s", versionId, resourceAttributes.Namespace)
+		pipelineId = pipelineVersion.PipelineId
 	}
-	return nil
+	return s.canAccessPipeline(ctx, pipelineId, resourceAttributes)
 }
 
 // Checks if a user can access a pipeline.
+// Adds parent namespace of pipeline id is not empty,
+// API group, version, and resource type.
 func (s *PipelineServer) canAccessPipeline(ctx context.Context, pipelineId string, resourceAttributes *authorizationv1.ResourceAttributes) error {
 	if !common.IsMultiUserMode() {
 		// Skip authorization if not multi-user mode.
 		return nil
 	}
 	if pipelineId != "" {
-		namespace, err := s.resourceManager.FetchNamespaceFromPipelineId(pipelineId)
+		pipeline, err := s.resourceManager.GetPipeline(pipelineId)
 		if err != nil {
 			return util.Wrapf(err, "Failed to access pipeline %s. Check if it exists and have a namespace assigned", pipelineId)
 		}
-		// Allow users to access pipelines in the default namespace.
-		if s.resourceManager.IsEmptyNamespace(namespace) {
-			return nil
+		resourceAttributes.Namespace = pipeline.Namespace
+		if len(resourceAttributes.Name) == 0 {
+			resourceAttributes.Name = pipeline.Name
```

Comment:
Why this change? Is `len(resourceAttributes.Name) == 0` a legit case, and expected to be patched here?

Reply:
If the `pipelineId` is not empty, the request targets a specific resource. Hence we add the name if it's missing. This is to avoid multiple calls to `GetRun`. Is some cases, like `GetPipelineByNameV1` we already have a name.